### PR TITLE
DOCSP-42324: Delete

### PR DESCRIPTION
**Pull request by mcmorisi**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-scala/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-42324>
Staging - <https://deploy-preview-65--docs-scala.netlify.app/write/delete/>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/write/delete.txt [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+-----------------
+
+You can perform delete operations in MongoDB by using the following methods:
+
+- ``deleteOne()``, which deletes *the first document* that matches the search criteria
+- ``deleteMany()``, which deletes *all documents* that match the search criteria
+
+Each delete method requires a **query filter** document, which specifies the
+search criteria that determine which documents to select for removal.
+To learn more about query filters, see the :ref:`scala-specify-query` guide.
```
Comment (lindseymoore) [inline source/write/delete.txt:57]:
```suggestion
To learn more about query filters, see the :ref:`scala-specify-query` guide.
```
Q: The ref link is broken. I see that the page has been completed, perhaps the ref isn't correct?

File: source/includes/write/delete.scala [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    val collection: MongoCollection[Document] = database.getCollection("restaurants")
+    // end-db-coll
+
+    // start-delete-one
+    val filter = equal("name", "Happy Garden")
+    val observable: Observable[DeleteResult] = collection.deleteOne(filter)
+
+    observable.subscribe(new Observer[DeleteResult] {
+      override def onNext(result: DeleteResult): Unit = println(s"Deleted document count: ${result.getDeletedCount}")
+      override def onError(e: Throwable): Unit = println(s"Error: $e")
```
Comment (lindseymoore) [inline source/includes/write/delete.scala:23]:
```suggestion
      override def onNext(result: DeleteResult): Unit = println("Deleted document count: ${result.getDeletedCount}")
      override def onError(e: Throwable): Unit = println("Error: $e")
```
Q: Not yet familiar with Scala, is the 's' necessary here in the print statements?

File: source/write/delete.txt [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+     - Description
+
+   * - ``collation()``
+     - | Specifies the kind of language collation to use when sorting
+         results. For more information, see :manual:`Collation </reference/collation/#std-label-collation>`
+         in the {+mdb-server+} manual.
+
+   * - ``hint()``
+     - | Specifies the index to use when matching documents.
+         For more information, see the :manual:`hint statement </reference/command/delete/#std-label-deletes-array-hint>`
```
Comment (lindseymoore) [inline source/write/delete.txt:129]:
```suggestion
         For more information, see the :manual:`hint field </reference/command/delete/#std-label-deletes-array-hint>`description
```
S: Confused by the use of "statement", as the link leads to field descriptions. Suggested alternative wording, would apply to all of the following uses of "statement".

File: source/write/delete.txt [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+         expressions that don't reference document fields. For more information,
+         see the :manual:`let statement
+         </reference/command/delete/#std-label-delete-let-syntax>` in the
+         {+mdb-server+} manual.
+
+   * - ``comment()``
+     - | Sets a comment to attach to the operation. For more
+         information, see the :manual:`delete command
+         fields </reference/command/delete/#command-fields>` guide in the
+         {+mdb-server+} manual for more information.
```
Comment (lindseymoore) [inline source/write/delete.txt:149]:
```suggestion
         {+mdb-server+} manual.
```
Nit: redundant

File: source/write/delete.txt [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+         variables for the operation. Values must be constant or closed
+         expressions that don't reference document fields. For more information,
+         see the :manual:`let statement
+         </reference/command/delete/#std-label-delete-let-syntax>` in the
+         {+mdb-server+} manual.
+
+   * - ``comment()``
+     - | Sets a comment to attach to the operation. For more
+         information, see the :manual:`delete command
+         fields </reference/command/delete/#command-fields>` guide in the
```
Comment (lindseymoore) [inline source/write/delete.txt:148]:
```suggestion
         information, see the :manual:`Command
         Fields </reference/command/delete/#command-fields>` section in the ``delete`` reference page of the
```
S: To clarify where the link is going.

File: source/write/delete.txt [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+The ``deleteOne()`` and ``deleteMany()`` methods each return a
+``DeleteResult`` instance. You can access the following information from
+a ``DeleteResult`` instance:
+
+- ``deletedCount``, which indicates the number of documents deleted
+- ``wasAcknowledged()``, which returns ``true`` if the server
+  acknowledges the result
+
+If the query filter does not match any documents, the driver doesn't delete any
+documents and the value of ``deletedCount`` is ``0``.
```
Comment (lindseymoore) [inline source/write/delete.txt:194]:
```suggestion
documents, and the value of ``deletedCount`` is ``0``.
```
comma needed

**Review by lindseymoore — CHANGES_REQUESTED** [review]
Small changes!

File: source/write/delete.txt (hunk 1 above)
Comment (mcmorisi) [reply source/write/delete.txt:57]: This PR was submitted last week, probably before the specify-query page was merged. This should be fine once this PR is merged.

File: source/includes/write/delete.scala (hunk 2 above)
Comment (mcmorisi) [reply source/includes/write/delete.scala:23]: This is a way to do string interpolation: [link to documentation](https://docs.scala-lang.org/scala3/book/string-interpolation.html)

File: source/write/delete.txt (hunk 3 above)
Comment (mcmorisi) [reply source/write/delete.txt:129]: Will make a change here

File: source/includes/write/delete.scala (hunk 2 above)
Comment (lindseymoore) [reply source/includes/write/delete.scala:23]: Thanks, looks good!

File: source/write/delete.txt [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``collation()``
+     - | Specifies the kind of language collation to use when sorting
+         results. For more information, see :manual:`Collation </reference/collation/#std-label-collation>`
+         in the {+mdb-server+} manual.
```
[the remark this answers is not in the document]
Comment (mcmorisi) [reply source/write/delete.txt:125]: I think the constant was created in a PR separate from this one. I can add it to the snooty.toml, but it should work fine once merged anyway.

**Review by jyemin — APPROVED** [review]

**Pull request merged**